File: src/main/java/com/example/stocksentry/dto/NewsResponse.java
package com.example.stocksentry.dto;

import lombok.Data;
import java.util.List;

@Data  // Lombok for getters/setters
public class NewsResponse {
    private String status;
    private int totalResults;
    private List<Article> articles;

    @Data
    public static class Article {
        private String title;
        private String description;
        private String content;
        private String publishedAt;  // publish date from API
        private String url;          // article URL


    }
}

File: app/src/main/java/com/stocksentry/app/data/models/NewsResponse.kt
package com.stocksentry.app.data.models

data class NewsResponse(
    val status: String? = null,
    val totalResults: Int = 0,
    val articles: List<Article> = emptyList()
)

data class Article(
    val title: String = "",
    val description: String = "",
    val content: String = "",
    val url: String = "",
    val publishedAt: String = "",
    val sourceUrl: String = ""
)

data class Source(
    val name: String = "",
    val id: String? = null
)

